testing_utils_test.go

```diff
@@ -165,6 +152,52 @@ func newFakeMarathonEndpoint(t *testing.T, configs *ConfigContainer) *endpoint {
 	}
 }
 
+// basicAuthMiddleware handles basic auth
+func basicAuthMiddleware(config *configContainer, next http.HandlerFunc) func(http.ResponseWriter, *http.Request) {
+	unauthorized := "{\"message\": \"invalid username or password\"}"
+
+	return func(w http.ResponseWriter, r *http.Request) {
+		// step: is authentication required?
+		if config.username != "" && config.password != "" {
+			u, p, found := r.BasicAuth()
+			// step: if no auth found, error it
+			if !found {
+				http.Error(w, unauthorized, 401)
+				return
+			}
+			// step: if username and password don't match, error it
+			if config.username != u || config.password != p {
+				http.Error(w, unauthorized, 401)
+				return
+			}
+		}
+
+		next(w, r)
+	}
+}
+
+// newFakeMarathonResponses reads in the marathon fake responses from the yaml file
+func newFakeMarathonResponses(t *testing.T) {
```

Comment:
problem with not passing the *testing.T is Once.Do() doesn't return an error ...